plugins/flytekit-kf-pytorch/flytekitplugins/kfpytorch/task.py

```diff
@@ -29,6 +31,36 @@ class PyTorch(object):
     num_workers: int
 
 
+@dataclass
+class Elastic(object):
+    """
+    Configuration for `torch elastic training <https://pytorch.org/docs/stable/elastic/run.html>`_.
+
+    Use this to run single- or multi-node distributed pytorch elastic training on k8s.
+
+    Single-node elastic training is executed in a k8s pod when `replicas` is set to 1.
+    Multi-node training is executed otherwise using a `Pytorch Job <https://github.com/kubeflow/training-operator>`_.
+
+    Args:
+        replicas int: Number of nodes
+        min_replicas int: Lower limit for the number of replicas to which the training job can scale down
+        max_replicas int: Upper limit for the number of replicas to which the training job can scale up.
+            Cannot be smaller than min_replicas.
+        nproc_per_node (Union[int, str]): Number of workers per node. Supported values are [auto, cpu, gpu, int].
+        start_method (str): Multiprocessing start method to use when creating workers.
+        monitor_interval (int): Interval, in seconds, to monitor the state of workers.
+        max_restarts (int): Maximum number of worker group restarts before failing.
+    """
+
+    replicas: int = 1
```

Comment:
can we just get rid of replicas?
and just make num_workers = max_replicas?
is that not right?
is replicas supposed to be desired_replicas?

Reply:
@fg91 I seem to agree. but I was thinking instead of nnodes keep min_replica and max_replica? Why not? though I will go with whatever you decide.
I actually am not really sure if HPA works with gloo correctly. Its not really easy to change the membership and work correctly. lets reserve this for later. Lets get the basic version out. If you make the change I can +1 tonight and merge. I have been testing on single node and it seems to work great.

Reply:
I have a slight preference for `nnodes=1 / "1:2"` over `min_replicas`, `max_replicas` for the following reasons:

* Not introduce a 3rd way in addition to `torchrun`'s `nnodes` and Kubeflow's `replicas`, `minReplicas`, `maxReplicas` that neither user knows
* I think most users will want to have a fixed number of workers but use `torchrun` simply because either a project like Alpacca or a library like ignite assumes it or because they want to do distributed training on a single node without operator.  It should be most simple for those users who don't think about min/max and I feel that `nnodes=2` is simpler for them than `min_replicas=2`, `max_replicas=2`.

If it's ok for you I will change back to `nnodes` and ping you again for review :)

Reply:
https://github.com/flyteorg/flytekit/pull/1603/commits/5552413312f6a662cec13661c75befebb4d2497c